src/super_gradients/common/crash_handler/crash_tips.py

```diff
@@ -0,0 +1,209 @@
+        factory_name, params_dict = RecipeFactoryFormatTip._get_factory_with_params(exc_value)
+
+        params_in_yaml = "\n".join(f"  {k}: {v}" for k, v in params_dict.items())
+        user_yaml = f"- {factory_name}:\n" + params_in_yaml
+        user_yaml = fmt_txt(user_yaml, indent=4, color="red")
+
+        correct_yaml = f"- {factory_name}:\n" + indent_string(params_in_yaml, indent_size=2)
+        correct_yaml = fmt_txt(correct_yaml, indent=4, color="green")
+        msg = "If your yaml looks like this:\n" f"{user_yaml}\n" f"Then you change your recipe to this:\n" f"{correct_yaml}"
+
+        return msg
+
+    @staticmethod
+    def _get_factory_with_params(exc_value: Exception) -> Tuple[str, dict]:
+        """Utility function to extract useful features from the exception.
+        :return: Name of the factory that (we assume) was not correctly defined
+        :return: Parameters that are passed to that factory
+        """
+        description = str(exc_value)
+        params_dict = re.search(r"received: (.*?)$", description).group(1)
+        params_dict = json_str_to_dict(params_dict)
+        factory_name = next(iter(params_dict))
+        params_dict.pop(factory_name)
+        return factory_name, params_dict
+
+
+class DDPNotInitializedTip(CrashTip):
+    """Note: I think that this should be caught within the code instead"""
+
+    @classmethod
+    def is_relevant(cls, exc_type: type, exc_value: Exception, exc_traceback: TracebackType):
+        expected_str = "Default process group has not been initialized, please make sure to call init_process_group."
+        return isinstance(exc_value, RuntimeError) and expected_str in str(exc_value)
+
+    @classmethod
+    def _get_explanation(cls, exc_type: type, exc_value: Exception, exc_traceback: TracebackType) -> str:
+        msg = "Your environment was not setup correctly for DDP."
+        return msg
+
+    @classmethod
+    def _get_solution(cls, exc_type: type, exc_value: Exception, exc_traceback: TracebackType) -> str:
+        msg = (
+            "Please run at the beginning of your script:\n"
+            f">>> {fmt_txt('from super_gradients.training.utils.distributed_training_utils import setup_gpu_mode', color='green')}\n"
+            f">>> {fmt_txt('from super_gradients.common.data_types.enum import MultiGPUMode', color='green')}\n"
+            f">>> {fmt_txt('setup_gpu_mode(gpu_mode=MultiGPUMode.DISTRIBUTED_DATA_PARALLEL, num_gpus=...)', color='green')}"
+        )
+        return msg
+
+
+# /!\ Only the CrashTips classes listed below will be used !! /!\
+ALL_CRASH_TIPS: List[Type[CrashTip]] = [TorchCudaMissingTip, RecipeFactoryFormatTip, DDPNotInitializedTip]
+
+
+def get_relevant_crash_tip_message(exc_type: type, exc_value: Exception, exc_traceback: TracebackType) -> Union[None, str]:
+    """Get a CrashTip class if relevant for input exception"""
+    for crash_tip in ALL_CRASH_TIPS:
+        if crash_tip.is_relevant(exc_type, exc_value, exc_traceback):
```

Comment:
What if we have more than one relevant tip? Maybe somehow account for such case and show multiple explanations/solutions.

Reply:
Good point, I think we should have 1 class per error, but with >=1 "tip"., each "tip" including 1 couple. of explanation/soluton.

Reply:
I added option to show multiple explanations/solutions here: 8f1cc5d28b908456ea4accb8e65e4a66a267abbc
Basically, explanation and solution were merged, and now _get_tips() -> List[str] needs to be implemented for each CrashTip class